### PR TITLE
Update variables.tf

### PR DESCRIPTION
**Pull request by joealewine**
### Description

Update variables.tf for Core Security Services DA.

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: solutions/standard/variables.tf [hunk 1]
```diff
@@ -54,64 +54,64 @@ variable "existing_secrets_endpoint_type" {
...
   type        = list(any)
-  description = "The list of resource tags that you want to associate with your Secrets Manager instance."
+  description = "The list of resource tags you want to associate with your Secrets Manager instance."
   default     = []
 }
 
 variable "public_engine_enabled" {
   type        = bool
-  description = "Set this to true to to configure an IBM Secrets Manager public certificate engine for an existing IBM Secrets Manager instance. If set to false, no public certificate engine will be configured for your secrets manager instance. For more details, see https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-prepare-order-certificates."
+  description = "Set this to true to configure an Secrets Manager public certificate engine for an existing Secrets Manager instance. If set to false, no public certificate engine will be configured for your instance."
```
Comment (alex-reiff) [inline solutions/standard/variables.tf:89]: an Secrets Manager -> a

File: solutions/standard/variables.tf [hunk 2]
```diff
@@ -183,13 +183,13 @@ variable "iam_engine_name" {
...
   type        = bool
-  description = "Set to true to skip the creation of an IAM authorization policy that permits all Secrets Manager instances in the resource group to read the encryption key from the KMS instance. If set to false, pass in a value for the KMS instance in the existing_kms_instance_crn variable."
+  description = "Set to true to skip the creation of an IAM authorization policy that permits all Secrets Manager instances in the resource group to read the encryption key. If set to false, pass in a value for the Key Protect or Hyper Protect Crypto Service instance in the existing_kms_instance_crn variable."
   default     = false
 }
 
 variable "existing_secrets_manager_kms_key_crn" {
   type        = string
-  description = "The CRN of an existing KMS key to use for Secrets Manager. If not supplied, a new key ring and key will be created."
+  description = "The CRN of an existing Key Protect key to use for Secrets Manager. If not supplied, a new key ring and key will be created."
```
Comment (alex-reiff) [inline solutions/standard/variables.tf:192]: can also be a Hyper Protect Crypto Services key

File: solutions/standard/variables.tf [hunk 3]
```diff
@@ -16,19 +16,19 @@ variable "use_existing_resource_group" {
 
 variable "resource_group_name" {
   type        = string
-  description = "The name of a new or an existing resource group in which to provision Secrets Manager resources to. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'. Optional if existing_secrets_manager_crn is not set."
+  description = "The name of a new or an existing resource group where the Secrets Manager resources are provisioned. If prefix input variable is passed then it is prefixed in front of the value in the format of '<prefix>-value'. Optional if existing_secrets_manager_crn is not set."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:19]:
```suggestion
  description = "The name of a new or existing resource group to provision resources to. If a prefix input variable is passed, it is added to the value in the `<prefix>-value` format. Optional if `existing_secrets_manager_crn` is not specified."
```

File: solutions/standard/variables.tf [hunk 4]
```diff
@@ -16,19 +16,19 @@ variable "use_existing_resource_group" {
...
   default     = null
 }
 
 variable "region" {
   type        = string
-  description = "The region in which to provision Secrets Manager resources."
+  description = "The region where Secrets Manager resources are provisioned."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:25]:
```suggestion
  description = "The region to provision resources to."
```

File: solutions/standard/variables.tf [hunk 5]
```diff
@@ -16,19 +16,19 @@ variable "use_existing_resource_group" {
...
   default     = "us-south"
 }
 
 variable "prefix" {
   type        = string
-  description = "(Optional) Prefix to append to all resources created by this solution."
+  description = "An optional prefix to append to all resources created by this solution."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:31]:
```suggestion
  description = "The prefix to apply to all resources created by this solution."
```

Trying to remove the optional label. It's clear in the Cloud UI.

File: solutions/standard/variables.tf [hunk 6]
```diff
@@ -38,13 +38,13 @@ variable "prefix" {
 
 variable "secrets_manager_instance_name" {
   type        = string
-  description = "The name to give the Secrets Manager instance that will be provisioned by this solution. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'"
+  description = "The name to give the Secrets Manager instance provisioned by this solution. If the prefix input variable is passed then it is prefixed infront of the value in the format of '<prefix>-value'"
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:41]:
```suggestion
  description = "The name to give the Secrets Manager instance provisioned by this solution. If a prefix input variable is specified, it is added to the value in the `<prefix>-value` format."
```

File: solutions/standard/variables.tf [hunk 7]
```diff
@@ -54,64 +54,64 @@ variable "existing_secrets_endpoint_type" {
   default     = "private"
   validation {
     condition     = contains(["public", "private"], var.existing_secrets_endpoint_type)
-    error_message = "Allowed values for 'existing_secrets_endpoint_type' are \"public\" and \"private\"."
+    error_message = "Only \"public\" and \"private\" are allowed values for 'existing_secrets_endpoint_type'."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:57]: We want to add the information to the description before they get the error (line 53)

description = "The endpoint type to use if `existing_secrets_manager_crn` is specified. Possible values: `public`, `private`."

File: solutions/standard/variables.tf [hunk 8]
```diff
@@ -54,64 +54,64 @@ variable "existing_secrets_endpoint_type" {
...
   }
 }
 
 variable "service_plan" {
   type        = string
-  description = "The service/pricing plan to use when provisioning a new Secrets Manager instance. Allowed values: 'standard' and 'trial'. Only used if `provision_sm_instance` is set to true."
+  description = "The pricing plan to use when provisioning a new Secrets Manager instance. Only 'standard' and 'trial' are allowed values. This variable is only used if `provision_sm_instance` is set to true."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:63]:
```suggestion
  description = "The pricing plan to use when provisioning a Secrets Manager instance. Possible values: `standard`, `trial`. Applies only if `provision_sm_instance` is set to `true`."
```

File: solutions/standard/variables.tf [hunk 9]
```diff
@@ -54,64 +54,64 @@ variable "existing_secrets_endpoint_type" {
...
     condition     = contains(["standard", "trial"], var.service_plan)
-    error_message = "Allowed values for sm_service_plan are \"standard\" and \"trial\"."
+    error_message = "Only \"standard\" and \"trial\" are allowed values for sm_service_plan."
   }
 }
 
 variable "allowed_network" {
   type        = string
-  description = "The types of service endpoints to set on the Secrets Manager instance. Possible values are `private-only` or `public-and-private`."
+  description = "The types of service endpoints to set on the Secrets Manager instance. Only `private-only` or `public-and-private` are allowed."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:73]:
```suggestion
  description = "The types of service endpoints to set on the Secrets Manager instance. Possible values: `private-only`, `public-and-private`."
```

File: solutions/standard/variables.tf [hunk 10]
```diff
@@ -54,64 +54,64 @@ variable "existing_secrets_endpoint_type" {
...
   type        = string
-  description = "Cloud Internet Service ID"
+  description = "Cloud Internet Service ID."
   default     = null
 }
 
 variable "dns_provider_name" {
   type        = string
-  description = "Name of the DNS provider for the public_cert secrets engine"
+  description = "The name of the DNS provider for the public_cert secrets engine."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:108]:
```suggestion
  description = "The name of the DNS provider for the public certificate secrets engine configuration."
```

File: solutions/standard/variables.tf (hunk 2 above)
Comment (SirSpidey) [inline solutions/standard/variables.tf:192]:
```suggestion
  description = "The CRN of a Key Protect or Hyper Protect Crypto Services key to use for Secrets Manager. If not specified, a key ring and key are created."
```

File: solutions/standard/variables.tf [hunk 11]
```diff
@@ -200,12 +200,12 @@ variable "existing_secrets_manager_kms_key_crn" {
 variable "existing_kms_instance_crn" {
   type        = string
   default     = null
-  description = "The CRN of the existed Hyper Protect Crypto Services or Key Protect instance. Only required if not supplying an existing KMS key to use for Secrets Manager."
+  description = "The CRN of the existing Hyper Protect Crypto Services or Key Protect instance. This is only required if not supplying an existing key to use for Secrets Manager."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:203]:
```suggestion
  description = "The CRN of the Hyper Protect Crypto Services or Key Protect instance. Applies only if `existing_secrets_manager_kms_key_crn` is not specified."
```

File: solutions/standard/variables.tf [hunk 12]
```diff
@@ -200,12 +200,12 @@ variable "existing_secrets_manager_kms_key_crn" {
...
 }
 
 variable "kms_endpoint_type" {
   type        = string
-  description = "The type of endpoint to be used for communicating with the KMS instance. Allowed values are: 'public' or 'private' (default). Only required if not supplying an existing KMS root key CRN."
+  description = "The type of endpoint to be used for communicating with the Key Protect or Hyper Protect Crypto Services instance. Both 'public' or 'private' are allowed. This value is only required if not supplying an existing root key CRN."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:208]:
```suggestion
  description = "The type of endpoint to use for communicating with the Key Protect or Hyper Protect Crypto Services instance. Possible values: `public`, `private`. Applies only if `existing_secrets_manager_kms_key_crn` is not specified."
```

File: solutions/standard/variables.tf [hunk 13]
```diff
@@ -216,13 +216,13 @@ variable "kms_endpoint_type" {
 variable "kms_key_ring_name" {
   type        = string
   default     = "sm-cos-key-ring"
-  description = "The name to give to the new KMS key ring that will be used to store the KMS key to enable advanced, customer-managed encryption for your Secrets Manager secrets. Only required if not supplying an existing KMS root key CRN. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'"
+  description = "The name to give to the new key ring used to store the key. Only required if not supplying an existing root key CRN. If the prefix input variable is passed it is added in the format of '<prefix>-value'."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:219]:
```suggestion
  description = "The name for the new key ring to store the key. Applies only if `existing_secrets_manager_kms_key_crn` is not specified. If a prefix input variable is passed, it is added to the value in the `<prefix>-value` format. ."
```

File: solutions/standard/variables.tf [hunk 14]
```diff
@@ -216,13 +216,13 @@ variable "kms_endpoint_type" {
...
 }
 
 variable "kms_key_name" {
   type        = string
   default     = "sm-cos-key"
-  description = "The name to give to the new KMS root key that will be used to enable advanced, customer-managed encryption for your Secrets Manager secrets. Only required if not supplying an existing KMS root key CRN. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'"
+  description = "The name to give to the new root key. Only required if not supplying an existing root key CRN. If the prefix input variable is passed it is added in the format of '<prefix>-value'."
```
Comment (SirSpidey) [inline solutions/standard/variables.tf:225]:
```suggestion
  description = "The name for the new root key. Applies only if `existing_secrets_manager_kms_key_crn` is not specified. If a prefix input variable is passed, it is added to the value in the `<prefix>-value` format."
```

**Review by SirSpidey — CHANGES_REQUESTED** [review]
Joe, suggestions from other DAs and from guidance I've come up with. Let me know if you have concerns with any of these.

**Comment by joealewine** [comment]
Thanks @alex-reiff and @SirSpidey for the great comments. I've either addressed or accepted them all.

File: solutions/standard/variables.tf (hunk 8 above)
Comment (SirSpidey) [reply solutions/standard/variables.tf:63]: open suggestion

File: solutions/standard/variables.tf (hunk 9 above)
Comment (SirSpidey) [reply solutions/standard/variables.tf:73]: open suggestion

File: solutions/standard/variables.tf (hunk 10 above)
Comment (SirSpidey) [reply solutions/standard/variables.tf:108]: open suggestion

**Review by SirSpidey — CHANGES_REQUESTED** [review]
@joealewine a few open suggestions

**Review by SirSpidey — APPROVED** [review]
👍

**Pull request merged**